src/middleware/PostQuery.ts

```diff
@@ -0,0 +1,67 @@
+import { Request, Response, NextFunction } from 'express';
+
+/*
+ *  This class will be for handling the transfer of the res.locals.graphqlGate object
+ *  to the webapp's backend
+ */
+
+export default class PostQuery {
+    private gateURI: string;
+
+    private projectID: string;
+
+    constructor(gateURI: string, projectID: string) {
+        this.gateURI = gateURI;
+        this.projectID = projectID;
+    }
+
+    // takes data from res.locals.graphqlGate and posts to webapp backend
+    public async post(req: Request, res: Response, next: NextFunction) {
+        if (!res.locals.graphqlGate)
+            throw new ReferenceError(
+                `gate-logger must be in the middleware chain right before graphql-gate 
+                \n ex. app.use(gate-logger())
+                \n app.use(graphQL-gate())`
+            );
+
+        const { complexity, depth, timestamp: time } = res.locals.graphqlGate;
+
+        const headers = {
+            'content-type': 'application/json',
+        };
+
+        // expecting variables to have projectID, complexity, depth, & timestamp properties
+        const variables = {
+            complexity,
+            depth,
+            time,
+            projectID: this.projectID,
+        };
+
+        const graphqlQuery = {
```

Comment:
This ties into my comments on the app but why not just use mongoose to log this data to the DB directly?

Reply:
Discussed

Reply:
What is the result of this discussion?